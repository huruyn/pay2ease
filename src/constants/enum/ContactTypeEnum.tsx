// eslint-disable-next-line import/prefer-default-export
export enum ContactTypeEnum {
  PERSONAL = 'personal',
  BUSINESS = 'business',
}
