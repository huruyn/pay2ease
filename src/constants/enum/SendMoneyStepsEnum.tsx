// eslint-disable-next-line import/prefer-default-export
export enum SendMoneyStepsEnum {
  TRANSACTION_START = 'transaction-start',
  TRANSACTION_DETAILS = 'transaction-details',
  TRANSACTION_SECURITY_QUESTIONS = 'transaction-security-questions',
  TRANSACTION_STATUS = 'transaction-status',
  TRANSACTION_COMPLETE = 'transaction-complete',
}
