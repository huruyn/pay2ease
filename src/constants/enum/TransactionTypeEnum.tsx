// eslint-disable-next-line import/prefer-default-export
export enum TransactionTypeEnum {
  SEND = 'send',
  REQUEST = 'request',
}
