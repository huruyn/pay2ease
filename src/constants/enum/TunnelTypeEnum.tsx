// eslint-disable-next-line import/prefer-default-export
export enum TunnelTypeEnum {
  ZIPPY_CASH = 'Zippy.Cash',
  INTERAC_E_TRANSFER = 'InteracÂ® e-Transfer',
}
