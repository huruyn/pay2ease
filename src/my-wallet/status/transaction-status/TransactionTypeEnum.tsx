// eslint-disable-next-line import/prefer-default-export
export enum TransactionTypeEnum {
  SENT = 'sent',
  RECEIVED = 'received',
  REQUESTED = 'requested',
}
