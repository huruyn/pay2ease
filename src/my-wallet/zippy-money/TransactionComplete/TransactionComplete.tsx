import {
  Button,
  Form,
  FormControl,
} from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import PageContainer from '../../../common/PageContainer';
import { TransactionInterface } from '../../../constants/interface/TransactionInterface';

export default function TransactionComplete({
  transactionType,
}: Pick<TransactionInterface, 'transactionType'>) {
  const navigate = useNavigate();

  return (
    <>
      <PageContainer
        title="Personal Account"
        subTitle="Made Fun With Zippy!"
        backdropImage="backdrop-image-2"
      >
        <div className="zippy-btn-group btn-group" role="group" aria-label="Zippy Cash">
          <Button className="active">
            Send Money
          </Button>
          <Button>
            Request Money
          </Button>
        </div>
        <div className="account-identifier">
          <FormControl
            placeholder="Enter @ZippyUsername or Email"
          />
          <i className="zippy-cash-icon zc-add" />
        </div>
        <FormControl
          className="amount"
          value="0.00"
          type="number"
          step=".01"
        />
        <div className="select-method">
          <Form.Check
            id="zippy-cash"
            type="radio"
            className="actions"
            label="Zippy.Cash"
            name="actions2"
            checked
          />
          <Form.Check
            id="interac-e-transfer"
            type="radio"
            className="actions"
            label="InteracÂ® e-Transfer"
            name="actions2"
          />
        </div>
        <div className="action">
          <Button
            className="zippy-btn"
            onClick={() => navigate(`/my-wallet/${transactionType}/transaction-start`)}
          >
            Zipp It
          </Button>
        </div>
      </PageContainer>
    </>
  );
}
