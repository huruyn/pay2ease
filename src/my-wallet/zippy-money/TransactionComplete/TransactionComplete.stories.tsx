import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import TransactionComplete from './TransactionComplete';
import { TransactionTypeEnum } from '../../../constants/enum/TransactionTypeEnum';

export default {
  title: 'my-wallet/zippy-money/TransactionComplete',
  component: TransactionComplete,
} as ComponentMeta<typeof TransactionComplete>;

const Template: ComponentStory<typeof TransactionComplete> = () => (
  <BrowserRouter>
    <TransactionComplete transactionType={TransactionTypeEnum.SEND} />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};

export const APIError = Template.bind({});
APIError.args = {};
APIError.parameters = {};
