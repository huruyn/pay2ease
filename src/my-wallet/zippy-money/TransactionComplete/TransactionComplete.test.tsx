import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TransactionComplete from './TransactionComplete';
import { TransactionTypeEnum } from '../../../constants/enum/TransactionTypeEnum';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useParams: () => ({
    stepId: 'details',
  }),
  useNavigate: () => mockedUsedNavigate,
}));

describe('TransactionComplete Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <TransactionComplete transactionType={TransactionTypeEnum.SEND} />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
