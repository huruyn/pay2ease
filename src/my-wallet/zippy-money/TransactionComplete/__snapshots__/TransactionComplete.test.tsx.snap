// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionComplete Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop backdrop-image-2"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Personal Account
        </h1>
        <h2
          className="catch"
        >
          Made Fun With Zippy!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          aria-label="Zippy Cash"
          className="zippy-btn-group btn-group"
          role="group"
        >
          <button
            className="active btn btn-primary"
            disabled={false}
            type="button"
          >
            Send Money
          </button>
          <button
            className="btn btn-primary"
            disabled={false}
            type="button"
          >
            Request Money
          </button>
        </div>
        <div
          className="account-identifier"
        >
          <input
            className="form-control"
            placeholder="Enter @ZippyUsername or Email"
          />
          <i
            className="zippy-cash-icon zc-add"
          />
        </div>
        <input
          className="amount form-control"
          step=".01"
          type="number"
          value="0.00"
        />
        <div
          className="select-method"
        >
          <div
            className="actions form-check"
          >
            <input
              checked={true}
              className="form-check-input"
              disabled={false}
              id="zippy-cash"
              name="actions2"
              type="radio"
            />
            <label
              className="form-check-label"
              htmlFor="zippy-cash"
              title=""
            >
              Zippy.Cash
            </label>
          </div>
          <div
            className="actions form-check"
          >
            <input
              className="form-check-input"
              disabled={false}
              id="interac-e-transfer"
              name="actions2"
              type="radio"
            />
            <label
              className="form-check-label"
              htmlFor="interac-e-transfer"
              title=""
            >
              InteracÂ® e-Transfer
            </label>
          </div>
        </div>
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Zipp It
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
